### ISSUE TITLE
[nnc][perf] Transcendental functions aren't properly vectorized on CPU

### ISSUE THREAD

**Issue by bertmaher** [pytorch/pytorch#50846]
## 🐛 Bug

The CPU fuser isn't vectorizing transcendental functions.  The root cause is that NNC expects to use Sleef, but the file including sleef in the fuser (torch/csrc/jit/tensorexpr/llvm_jit.cpp) doesn't define __AVX__ so the vectorized functions aren't defined.  We need to use the same tricks as eager mode to compile-in sleef, but detect the proper instruction set support at runtime using cpuinfo.

## To Reproduce

```
import timeit
import torch

torch._C._jit_override_can_fuse_on_cpu(True)                                                                                                                                                   
torch._C._debug_set_fusion_group_inlining(False)                                                                                                                                               

x = torch.randn(1024, 1024)

@torch.jit.script                                                                                                                                                                              
def script_tanh(x):
    return torch.tanh(x)

for _ in range(5):
    script_tanh(x)
    
print(timeit.timeit(stmt="torch.tanh_(x)", globals=globals(), number=1000))                                                                                                                    
print(timeit.timeit(stmt="script_tanh(x)", globals=globals(), number=1000))                                                                                                                    
```

Results on my 2.4 GHz 24-core Broadwell show about 5x slowdown for the fuser.
```
2.983580728061497
15.662288276012987
```

The problem is that it's dispatching to libm's scalar tanhf, whereas the eager version uses sleef or VML.  (VML is faster than sleef, but harder to use in the fuser).

## Expected behavior

Perf at least on par with sleef.

**Comment by cheng-chang** [comment]
If set 
```
from caffe2.python import core
core.GlobalInit(["python", "--torch_jit_llvm_use_fast_intrinsics", "1"])
```
, then the performance is on par.

**Comment by bertmaher** [comment]
Just to be clear that `GlobalInit` thing is kind of a developer trick to let us experiment with approximations, not anything stable.

**Comment by bertmaher** [comment]
Fixed by #51190
